scripts/build.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+# This is a generic build script for the GDK for Unity.
+# The expected usage is as follows:
+#   bash build.sh <unity_project_dir> <worker_type> <build_target> <log_file (optional)>
+
+set -e -u -x -o pipefail
+
+source "$(dirname "$0")/scripts/pinned-tools.sh"
+source "$(dirname "$0")/scripts/profiling.sh"
+
+UNITY_PROJECT_DIR=$1
+WORKER_TYPE=$2
+BUILD_TARGET=$3
+LOG_FILE=${4:-} # Optional argument - will log to console otherwise
+
+RUN_UNITY_PATH="$(dirname "$0")/tools/RunUnity/RunUnity.csproj"
+
+markStartOfBlock "Building ${WORKER_TYPE} for ${BUILD_TARGET}"
+
+pushd "${UNITY_PROJECT_DIR}"
+    dotnet run -p "${RUN_UNITY_PATH}" -- \
+        -projectPath "${UNITY_PROJECT_DIR}" \
+        -batchmode \
```

Comment:
do we want -nographics here as well?

Reply:
I don't think so - it can cause problems with building I think. cc @zeroZshadow

Reply:
It can screw up builds that do light baking and other stuff